src/Symfony/Component/Ldap/Security/LdapUser.php

```diff
@@ -95,4 +97,14 @@ public function isEqualTo(UserInterface $user): bool
 
         return true;
     }
+
+    public function __serialize(): array
+    {
+        return [$this->entry, $this->identifier, null, $this->roles, $this->extraFields];
+    }
+
+    public function __unserialize(array $data): void
+    {
+        [$this->entry, $this->identifier, $this->password, $this->roles, $this->extraFields] = $data;
+    }
```

Comment:
The password property should default to null in the constructor:
```diff
-private ?string $password
+private ?string $password = null
```

Then we can enhance the compatibility of the logic with existing payloads (the current update breaks loading existing payloads):

```suggestion
    public function __serialize(): array
    {
        $data = (array) $this;
        unset($data[\sprintf("\0%s\0password"), self::class]);

        return $data;
    }
```